sources/packages/backend/apps/api/src/services/student-file/student-file.service.ts

```diff
@@ -119,35 +119,78 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
     auditUserId: number,
     uniqueFileNames: string[],
     fileOrigin: FileOriginType,
-    groupName: string,
     options?: FileUploadOptions,
   ): Promise<UpdateResult> {
     let updateResult: UpdateResult;
-    await this.dataSource.transaction(async (transactionalEntityManager) => {
-      updateResult = await transactionalEntityManager
-        .getRepository(StudentFile)
-        .update(
-          {
-            student: { id: studentId },
-            uniqueFileName: In(uniqueFileNames),
-          },
-          {
-            groupName,
-            fileOrigin,
-            metadata: options?.metadata,
-            modifier: { id: auditUserId } as User,
-          },
-        );
+    if (options.entityManager) {
+      updateResult = await this.updateStudentFile(
+        options.entityManager,
+        studentId,
+        auditUserId,
+        uniqueFileNames,
+        fileOrigin,
+        options,
+      );
 
       // Create student notification on file upload.
       if (options?.saveFileUploadNotification) {
```

Comment:
Saving file upload notification can go inside the private method.

![image](https://github.com/bcgov/SIMS/assets/54600590/33e207b2-1e1e-4c00-91aa-018e4d853a33)